common/src/main/scala/notifications/providers/NotificationProvider.scala

```diff
@@ -0,0 +1,11 @@
+package notifications.providers
+
+import models.Push
+
+import scala.concurrent.Future
+import scalaz.\/
+
+trait NotificationProvider {
```

Comment:
*Provider suffix is unfortunate - it's the interface of NotificationSender not a provider